qiskit/providers/aer/backends/aer_compiler.py

```diff
@@ -40,27 +41,29 @@ def compile(self, circuits, basis_gates=None, optypes=None):
                             (default: None).
 
         Returns:
-            QuantumCircuit or list: QuantumCircuit (or list
-                of QuantumCircuit objects) without control-flow instructions
+            tuple: Typle of list QuantumCircuit without control-flow
+                   instructions and list of optypes for each circuit.
         """
-        if isinstance(circuits, QuantumCircuit):
+        if not isinstance(circuits, list):
             circuits = [circuits]
```

Comment:
The previous form was probably better - this way means that if you pass `circuits` as a tuple or any other iterable, it won't work, whereas previously it would (type-hint notwithstanding).